src/providers/storyblok/runtime.ts

```diff
@@ -0,0 +1,54 @@
+import type { ProviderGetImage } from 'src'
+
+const defaultModifiers = {
+  width: '0',
+  height: '0'
+}
+
+const buildWidth = (mergeModifiers) => {
+  const { width, height } = mergeModifiers
+
+  if (width === '0' && height === '0') { return '' }
+
+  return `/${width}x${height}`
+}
+
+const buildFit = ({ fit }) => fit ? `/fit-${fit}` : ''
+
+const buildSmart = ({ smart }) => typeof smart !== 'undefined' ? '/smart' : ''
+
+const buildFilters = (filters) => {
+  if (!filters) { return '' }
+
+  let string = '/filters'
+
+  for (const f in filters) {
+    string += `:${f}(${filters[f]})`
+  }
+
+  return string
+}
+
+const overrideUndefined = (modifiers) => {
+  const ret = modifiers
+  for (const key in modifiers) {
+    if (typeof modifiers[key] === 'undefined' && typeof defaultModifiers[key] !== 'undefined') {
+      ret[key] = defaultModifiers[key]
+    }
+  }
+  return ret
+}
+
+export const getImage: ProviderGetImage = (src, { modifiers = {}, baseURL = 'https://img2.storyblok.com' } = {}) => {
+  const mergeModifiers = overrideUndefined({
+    ...defaultModifiers,
+    ...modifiers
+  })
+
+  const srcWithoutBase = src.replace('https://a.storyblok.com', '')
```

Comment:
Is there a reason to unconditionally trim `a.storyblock.com` from src?

Reply:
Yes all urls from storyblok come like this and need to be change by the cdn url. https://www.storyblok.com/docs/image-service

Reply:
Thanks. Would you please add a comment above linking to docs for replacing?

Reply:
- we probably also need to ensure not adding duplicate prefix is src is img2

- Also other providers use resource address instead of full URL. This feature seems little bit inconsistency

Reply:
- Not sure to understand the first point
- Yes but it really depends on the provider :s or I have to transform it with methods before passing it to the component